src/llvm_backend_type.cpp

```diff
@@ -123,10 +124,16 @@ lbValue lb_get_type_info_ptr(lbModule *m, Type *type) {
 
 	lbValue res = {};
 	res.type = t_type_info_ptr;
-	res.value = LLVMConstGEP(lb_global_type_info_data_ptr(m).value, indices, cast(unsigned)gb_count_of(indices));
+	lbValue data = lb_global_type_info_data_ptr(m);
+	res.value = LLVMConstGEP2(lb_type(m, type_deref(data.type)), data.value, indices, cast(unsigned)gb_count_of(indices));
 	return res;
 }
 
+// The use of this method needs to be eliminated.
+LLVMTypeRef lb_llvm_get_pointer_type(LLVMTypeRef type)
```

Comment:
Why isn't this just called `llvm_get_element_type` rather than `LLVMGetElementType`?

Reply:
Because I want to separate the valid use of LLVMGetElementType (using it on arrays, vectors and matrices) vs the deprecated use (using it on pointers). So this one is the one I have to remove all uses of.

Reply:
I've already removed most use cases of it already!